### PR TITLE
Refactor Storage for TestResultCoordinator

### PR DESCRIPTION
**Pull request by dylanbronson**
Moving away from static and introducing DI to TRC.

### CODE REVIEW

File: test/modules/TestResultCoordinator/Controllers/TestOperationResultController.cs [hunk 1]
```diff
@@ -11,13 +10,20 @@ namespace TestResultCoordinator.Controllers
     [ApiController]
     public class TestOperationResultController : Controller
     {
+        private readonly ITestOperationResultStorage testOperationResultStorage;
```
Comment (varunpuranik) [inline test/modules/TestResultCoordinator/Controllers/TestOperationResultController.cs]: Nit: avoid private keyword for fields, it is the default

File: test/modules/TestResultCoordinator/Controllers/TestOperationResultController.cs [hunk 2]
```diff
@@ -11,13 +10,20 @@ namespace TestResultCoordinator.Controllers
...
+
+        public TestOperationResultController(ITestOperationResultStorage testOperationResultStorage)
+        {
+            this.testOperationResultStorage = testOperationResultStorage;
```
Comment (varunpuranik) [inline test/modules/TestResultCoordinator/Controllers/TestOperationResultController.cs]: check null

File: test/modules/TestResultCoordinator/ITestOperationResultStorage.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace TestResultCoordinator
+{
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Storage;
+    public interface ITestOperationResultStorage
+    {
+        ISequentialStore<TestOperationResult> GetStoreFromSource(string source);
+
+        Task<bool> AddResultAsync(TestOperationResult testOperationResult);
```
Comment (varunpuranik) [inline test/modules/TestResultCoordinator/ITestOperationResultStorage.cs]: How do you retrieve the results?

File: test/modules/TestResultCoordinator/ITestOperationResultStorage.cs [hunk 4]
```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace TestResultCoordinator
+{
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Storage;
+    public interface ITestOperationResultStorage
+    {
+        ISequentialStore<TestOperationResult> GetStoreFromSource(string source);
```
Comment (varunpuranik) [inline test/modules/TestResultCoordinator/ITestOperationResultStorage.cs]: What is this used for?

File: test/modules/TestResultCoordinator/PartitionReceiverHandler.cs [hunk 5]
```diff
@@ -19,11 +19,14 @@ class PartitionReceiveHandler : IPartitionReceiveHandler
...
         readonly string trackingId;
+        readonly ITestOperationResultStorage storage;
 
-        public PartitionReceiveHandler(string trackingId, string deviceId)
+        public PartitionReceiveHandler(string trackingId, string deviceId, ITestOperationResultStorage storage)
         {
             this.trackingId = Preconditions.CheckNonWhiteSpace(trackingId, nameof(trackingId));
             this.deviceId = Preconditions.CheckNonWhiteSpace(deviceId, nameof(deviceId));
+            this.storage = Preconditions.CheckNotNull(storage);
+            Logger.LogInformation($"registration worked: {storage.GetStoreFromSource("relayer1.send")}");
```
Comment (varunpuranik) [inline test/modules/TestResultCoordinator/PartitionReceiverHandler.cs]: ? You are getting the store only to log?

File: test/modules/TestResultCoordinator/TestOperationResultStorage.cs [hunk 6]
```diff
@@ -12,12 +12,23 @@ namespace TestResultCoordinator
...
 
-    class TestOperationResultStorage
+    public class TestOperationResultStorage : ITestOperationResultStorage
     {
         static readonly ILogger Logger = ModuleUtil.CreateLogger("TestResultCoordinator");
-        static Dictionary<string, ISequentialStore<TestOperationResult>> resultStores;
+        private Dictionary<string, ISequentialStore<TestOperationResult>> ResultStores { get; set; }
 
-        public static async Task InitAsync(string storagePath, ISystemEnvironment systemEnvironment, bool optimizeForPerformance, List<string> resultSources)
+        public static async Task<TestOperationResultStorage> CreateStorage(string storagePath, ISystemEnvironment systemEnvironment, bool optimizeForPerformance, List<string> resultSources)
```
Comment (varunpuranik) [inline test/modules/TestResultCoordinator/TestOperationResultStorage.cs]: Nit: suggest renaming to Create

File: test/modules/TestResultCoordinator/TestOperationResultStorage.cs [hunk 7]
```diff
@@ -12,12 +12,23 @@ namespace TestResultCoordinator
...
+        {
+            Dictionary<string, ISequentialStore<TestOperationResult>> resultStores = await TestOperationResultStorage.InitAsync(storagePath, systemEnvironment, optimizeForPerformance, resultSources);
```
Comment (varunpuranik) [inline test/modules/TestResultCoordinator/TestOperationResultStorage.cs]: Remove InitAsync. This is essentially your Init now.

File: test/modules/TestResultCoordinator/TestOperationResultStorage.cs (hunk 6 above)
Comment (varunpuranik) [inline test/modules/TestResultCoordinator/TestOperationResultStorage.cs]: We should pass in the DbStoreProvider... that way we can use RocksDb / InMemoryStore depending on config.

File: test/modules/TestResultCoordinator/ITestOperationResultStorage.cs (hunk 3 above)
Comment (dylanbronson) [reply test/modules/TestResultCoordinator/ITestOperationResultStorage.cs]: We use GetStoreFromSource with ReportMetadata, which contains the following info about the report: ExpectedSource, ActualSource, TestReportType, and TestOperationResultType. 

For each ReportType we have, we will have a new testResultReportGenerator that will generate a report for that Type.

GetStoreFromSource is used in TestReportGeneratorFactory to create TestResultReportGenerator's (currently only CountingReportGenerator)

File: test/modules/TestResultCoordinator/TestOperationResultStorage.cs (hunk 6 above)
Comment (dylanbronson) [reply test/modules/TestResultCoordinator/TestOperationResultStorage.cs]: Why would that be something we want configured in our tests? I can't think of why we wouldn't want to use persistent storage?

**Pull request closed without merging**